CollAction/Helpers/ProjectNormalizeStringExtensions.cs

```diff
@@ -0,0 +1,82 @@
+using System.Linq;
+
+namespace CollAction.Helpers
+{
+    public static class ProjectNormalizeStringExtensions
+    {
+
+        public static string ToUrlSlug(this string value){
+
+            //First to lower case
+            value = value.ToLowerInvariant();
+
+            //Replace spaces
+            value = Regex.Replace(value, @"\s", "-", RegexOptions.Compiled);
+
+            //Remove invalid chars
+            value = Regex.Replace(value, @"[^a-z0-9\s-_]", "",RegexOptions.Compiled);
+
+            //Trim dashes from end
+            value = value.Trim('-', '_');
+
+            //Replace double occurences of - or _
+            value = Regex.Replace(value, @"([-_]){2,}", "$1", RegexOptions.Compiled);
+
+            return value;
+        }
+    
+        public static string RemoveAccent(this string txt) 
+        { 
+            byte[] bytes = System.Text.Encoding.GetEncoding("Cyrillic").GetBytes(txt); 
+            return System.Text.Encoding.ASCII.GetString(bytes); 
+        }
+
+        public static string Translit(this string str)
+        {
+            string[] lat_up = {"A", "B", "V", "G", "D", "E", "Yo", "Zh", "Z", "I", "Y", "K", "L", "M", "N", "O", "P", "R", "S", "T", "U", "F", "Kh", "Ts", "Ch", "Sh", "Shch", "\"", "Y", "'", "E", "Yu", "Ya"};
+            string[] lat_low = {"a", "b", "v", "g", "d", "e", "yo", "zh", "z", "i", "y", "k", "l", "m", "n", "o", "p", "r", "s", "t", "u", "f", "kh", "ts", "ch", "sh", "shch", "\"", "y", "'", "e", "yu", "ya"};
+            string[] rus_up = {"А", "Б", "В", "Г", "Д", "Е", "Ё", "Ж", "З", "И", "Й", "К", "Л", "М", "Н", "О", "П", "Р", "С", "Т", "У", "Ф", "Х", "Ц", "Ч", "Ш", "Щ", "Ъ", "Ы", "Ь", "Э", "Ю", "Я"};
+            string[] rus_low = { "а", "б", "в", "г", "д", "е", "ё", "ж", "з", "и", "й", "к", "л", "м", "н", "о", "п", "р", "с", "т", "у", "ф", "х", "ц", "ч", "ш", "щ", "ъ", "ы", "ь", "э", "ю", "я"};
+            for (int i = 0; i <= 32; i++)
+            {
+                str = str.Replace(rus_up[i],lat_up[i]);
+                str = str.Replace(rus_low[i],lat_low[i]);              
+            }
+            return str;
+        }
+
+        
+        public static string RemoveDiacriticsFromString(this string text) 
+            {
+                var normalizedString = text.Normalize(NormalizationForm.FormD);
+                var stringBuilder = new StringBuilder();
+
+                foreach (var c in normalizedString)
+                {
+                    var unicodeCategory = CharUnicodeInfo.GetUnicodeCategory(c);
+                    if (unicodeCategory != UnicodeCategory.NonSpacingMark)
+                    {
+                        stringBuilder.Append(c);
+                    }
+                }
+
+                return stringBuilder.ToString().Normalize(NormalizationForm.FormC);
+            }
+        
+
+        public static string NormalizeUriPart(this string project)
+        {
+            return project
+            .Translit()
```

Comment:
Might need some indentation. Usually you indent it with 4 spaces.

Reply:
Fixed